File: examples/python/basic_task.py
"""
basic_task.py - 基础任务创建与执行示例
"""
import asyncio
import os
from honor_agent import HonorAgent
from honor_agent.models import Task, TaskParams


async def main():
    api_key = os.getenv("HONOR_API_KEY", "your-api-key")
    base_url = os.getenv("HONOR_BASE_URL", "http://localhost:8000")
    
    client = HonorAgent(api_key=api_key, base_url=base_url)
    
    task = await client.tasks.create(
        name="数据分析任务",
        description="分析销售数据并生成周报",
        agents=["data_analyst"],
        params=TaskParams(
            data_source="sales_db",
            date_range="last_week"
        )
    )
    
    print(f"✓ 任务创建成功: {task.id}")
    
    result = await client.tasks.run(task_id=task.id)
    
    print(f"✓ 执行状态: {result.status}")
    if result.output:
        print(f"✓ 输出: {result.output}")
    
    return result


if __name__ == "__main__":
    result = asyncio.run(main())

File: examples/javascript/basic-task.ts
/**
 * basic-task.ts - 基础任务创建与执行示例
 */
import { HonorAgent } from '@honor-agent/sdk';

async function main() {
  const client = new HonorAgent({
    apiKey: process.env.HONOR_API_KEY || 'your-api-key',
    baseUrl: process.env.HONOR_BASE_URL || 'http://localhost:8000',
  });

  try {
    const task = await client.tasks.create({
      name: '数据分析任务',
      description: '分析销售数据并生成周报',
      agents: ['data_analyst'],
      params: {
        dataSource: 'sales_db',
        dateRange: 'last_week',
      },
    });

    console.log(`✓ 任务创建成功: ${task.id}`);

    const result = await client.tasks.run({ taskId: task.id });

    console.log(`✓ 执行状态: ${result.status}`);
    if (result.output) {
      console.log(`✓ 输出:`, result.output);
    }

    return result;
  } catch (error) {
    console.error('任务执行失败:', error);
    throw error;
  }
}

main().catch(console.error);
